lib/galaxy/webapps/galaxy/api/item_tags.py

```diff
@@ -3,71 +3,112 @@
+                tags=[tagged_item_tag],
+                summary=f"Show tag based on {tagged_item_id}",
+            )
+            def show(
+                self,
+                trans: ProvidesAppContext = DependsOnTrans,
+                id: DecodedDatabaseIdField = Path(..., title="Item ID", description=f"{tagged_item_id}"),
+                tag_name: str = Path(..., title="Tag Name"),
+            ) -> ItemTagsResponse:
+                return self.manager.show(trans, tagged_item_class, id, tag_name)
+
+            @router.post(
+                f"/api/{prefix}/{{id}}/tags/{{tag_name}}",
+                tags=[tagged_item_tag],
+                summary=f"Create tag based on {tagged_item_id}",
+            )
+            def create(
+                self,
+                trans: ProvidesAppContext = DependsOnTrans,
+                id: DecodedDatabaseIdField = Path(..., title="Item ID", description=f"{tagged_item_id}"),
+                tag_name: str = Path(..., title="Tag Name"),
+                payload: ItemTagsCreatePayload = Body(...),
+            ) -> ItemTagsResponse:
+                return self.manager.create(trans, tagged_item_class, id, tag_name, payload)
+
+            @router.put(
+                f"/api/{prefix}/{{id}}/tags/{{tag_name}}",
+                tags=[tagged_item_tag],
+                summary=f"Update tag based on {tagged_item_id}",
+            )
+            def update(
+                self,
+                trans: ProvidesAppContext = DependsOnTrans,
+                id: DecodedDatabaseIdField = Path(..., title="Item ID", description=f"{tagged_item_id}"),
+                tag_name: str = Path(..., title="Tag Name"),
+                payload: ItemTagsCreatePayload = Body(...),
+            ) -> ItemTagsResponse:
+                return self.manager.create(trans, tagged_item_class, id, tag_name, payload)
+
+            @router.delete(
+                f"/api/{prefix}/{{id}}/tags/{{tag_name}}",
+                tags=[tagged_item_tag],
+                summary=f"Delete tag based on {tagged_item_id}",
+            )
+            def delete(
+                self,
+                trans: ProvidesAppContext = DependsOnTrans,
+                id: DecodedDatabaseIdField = Path(..., title="Item ID", description=f"{tagged_item_id}"),
+                tag_name: str = Path(..., title="Tag Name"),
+            ) -> bool:
+                return self.manager.delete(trans, tagged_item_class, id, tag_name)
+
+        return Temp
+
+
+prefixs = {
+    "histories": ["History", "history_id", "histories"],
+    "histories/{history_id}/contents": ["HistoryDatasetAssociation", "history_content_id", "histories"],
+    "workflows": ["StoredWorkflow", "workflow_id", "workflows"],
+}
+for prefix, tagged_item in prefixs.items():
+    tagged_item_class, tagged_item_id, tagged_item_tag = tagged_item
+    router.cbv(FastAPIItemTags.create_class(prefix, tagged_item_class, tagged_item_id, tagged_item_tag))
```

Comment:
We already have a TaggableItemClass:

https://github.com/arash77/galaxy/blob/a10f574c5692ef96d979d0a078ed35bc28e89614/lib/galaxy/managers/tags.py#L16

I wonder if we could use this to tighten the type of tagged_item_class and the model_class in the responses, but it seems this kind of API is only supported for these 3 classes (History, HistoryDatasetAssociation, and StoredWorkflow), and not for all the taggable elements, so it might not be worth it, just thinking out loud more than a request.

Reply:
I am confused, for what purpose should I use `TaggableItemClass`?

Reply:
What I meant is just that `tagged_item_class` would be of type `Any` and maybe we could improve the typing here, but I see that we support this kind of "extended tags API" only for 3 out of 7 taggable items so it may not be worth using it for now.
In summary, I just wanted to point out that small inconsistency in the current API, but it is fine as it is for now, we can try to expand or improve it later if necessary :)